pandas/core/base.py

```diff
@@ -990,13 +991,73 @@ def factorize(self, sort=False, na_sentinel=-1):
         from pandas.core.algorithms import factorize
         return factorize(self, sort=sort, na_sentinel=na_sentinel)
 
-    def searchsorted(self, key, side='left'):
-        """ np.ndarray searchsorted compat """
+    _shared_docs['searchsorted'] = (
+        """Find indices where elements should be inserted to maintain order.
 
+        Find the indices into a sorted %(klass)s `self` such that, if the
+        corresponding elements in `v` were inserted before the indices, the
+        order of `self` would be preserved.
+
+        Parameters
+        ----------
+        v : array_like
+            Values to insert into `a`.
+        side : {'left', 'right'}, optional
+            If 'left', the index of the first suitable location found is given.
+            If 'right', return the last such index.  If there is no suitable
+            index, return either 0 or N (where N is the length of `a`).
+
+        Returns
+        -------
+        indices : array of ints
+            Array of insertion points with the same shape as `v`.
+
+        See Also
+        --------
+        numpy.searchsorted
+
+        Notes
+        -----
+        Binary search is used to find the required insertion points.
+
+        Examples
+        --------
+        >>> x = pd.Series([1, 2, 3])
+        >>> x
+        0    1
+        1    2
+        2    3
+        dtype: int64
+        >>> x.searchsorted(4)
+        array([3])
+        >>> x.searchsorted([0, 4])
+        array([0, 3])
+        >>> x.searchsorted([1, 3], side='left')
+        array([0, 2])
+        >>> x.searchsorted([1, 3], side='right')
+        array([1, 3])
+        >>>
+        >>> x = pd.Categorical(['apple', 'bread', 'bread', 'cheese', 'milk' ])
+        [apple, bread, bread, cheese, milk]
+        Categories (4, object): [apple < bread < cheese < milk]
+        >>> x.searchsorted('bread')
+        array([1])     # Note: an array, not a scalar
+        >>> x.searchsorted(['bread'])
+        array([1])
+        >>> x.searchsorted(['bread', 'eggs'])
+        array([1, 4])
+        >>> x.searchsorted(['bread', 'eggs'], side='right')
+        array([3, 4])    # eggs before milk
+        """)
+
+    @Substitution(klass='IndexOpsMixin')
+    @Appender(_shared_docs['searchsorted'])
+    def searchsorted(self, v, side='left', *args):
         # FIXME in GH7447
-        # needs coercion on the key (DatetimeIndex does alreay)
+        # needs coercion on the key (DatetimeIndex does already)
         # needs tests/doc-string
-        return self.values.searchsorted(key, side=side)
+        com.validate_args(args, min_length=0, max_length=1)
```

Comment:
Fair enough, though I think the change would have to be more gradual, since if the issue I'm fixing is any indication, many people I suspect might believe `np.searchsorted` (and various other `numpy` functions) should "work" with `pandas` objects.  In fact, that is the idea behind several `pandas` functions that have counterparts in `numpy` as far as I have seen.

This may be another issue you might want to discuss with @jreback , since he has generally advocated a more `numpy` friendly interface for `pandas` so that more `numpy` dependent users can use `pandas` more easily given how similar the interface may appear at the surface.

At this point, IMO it would be best to save the `Exception` raising for a separate PR in which the interface would change across the board to become more `numpy`-hostile (for lack of a better term), as this does feel like a somewhat significant change.